src/main/java/com/company/Main.java

```diff
@@ -1,8 +1,21 @@
 package com.company;
 
+import java.io.IOException;
+import java.util.Scanner;
+
 public class Main {
 
-    public static void main(String[] args) {
-        System.out.println("Success");
+    public static void main(String[] args) throws IOException {
+        Client client = new Client(args[0]);
+        while (true) {
+            client.work();
+//            if(System.in.available() != -1) {
+//                Scanner scanner = new Scanner(System.in);
+//                String s = scanner.nextLine();
+//                if(s.equals("stop")) break;
+//            }
+            if (false) break;
+        }
+        client.close();
```

Comment:
Когда мы сюда попадём?

Reply:
Никогда, я это понимаю, но у меня не получилось заставить System.in ждать ввода и при этом чтобы клиент крутился на фоне, так как программа однопоточная.

Reply:
> т.е. сокет никогда не будет закрыт? пн, 28 сент. 2020 г. в 14:42, Nikita Drozdov <notifications@github.com>:
> […](#)
> ***@***.**** commented on this pull request. ------------------------------ In src/main/java/com/company/Main.java <[#2 (comment)](https://github.com/hater4u/MulticastCopy/pull/2#discussion_r495746451)>: > public class Main { - public static void main(String[] args) { - System.out.println("Success"); + public static void main(String[] args) throws IOException { + Client client = new Client(args[0]); + while (true) { + client.work(); +// if(System.in.available() != -1) { +// Scanner scanner = new Scanner(System.in); +// String s = scanner.nextLine(); +// if(s.equals("stop")) break; +// } + if (false) break; + } + client.close(); Никогда, я это понимаю, но у меня не получилось заставить System.in ждать ввода и при этом чтобы клиент крутился на фоне, так как программа однопоточная. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#2 (comment)](https://github.com/hater4u/MulticastCopy/pull/2#discussion_r495746451)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGF7ICYG5OZW7GKK64WQURLSIA47LANCNFSM4R2ZTLQA> .
> -- С уважением, Ковалев Тимофей.

Да, не будет, но я сейчас подумав понял, если обернуть этот цикл в блок try catch finally и в finally делать client.close то можно считать что сокет закрыт?